ee/hogai/session_summaries/utils.py

```diff
@@ -137,3 +139,21 @@ def unpack_full_event_id(full_event_id: str | None, session_id: str | None = Non
 def strip_raw_llm_content(raw_content: str) -> str:
     """Strip the first and the last line of the content to load the YAML data only into JSON"""
     return raw_content.strip("```yaml\n").strip("```").strip()  # noqa: B005
+
+
+@lru_cache(maxsize=128)
+def _get_encoding_for_model(model: str):
+    """Cached wrapper for tiktoken.encoding_for_model."""
+    return tiktoken.encoding_for_model(model)
```

Comment:
Looking at `get_encoding()` implementation, that seems to already cache the encodings constructed, no?

Reply:
Good point, removed.